docs/topics/numbers.md

```diff
@@ -0,0 +1,289 @@
+
+For real numbers, Kotlin provides floating-point types `Float` and `Double`.
+According to the [IEEE 754 standard](https://en.wikipedia.org/wiki/IEEE_754),
+floating point types differ by their _decimal place_, that is, how many decimal digits they can store.
+`Float` reflects the IEEE 754 _single precision_, while `Double` provides _double precision_.
+
+| Type	 |Size (bits)|Significant bits|Exponent bits|Decimal digits|
+|--------|-----------|--------------- |-------------|--------------|
+| Float	 | 32        |24              |8            |6-7            |
+| Double | 64        |53              |11           |15-16          |    
+
+You can initialize `Double` and `Float` variables  with numbers having a fractional part.
+It's separated from the integer part by a period (`.`)
+For variables initialized with fractional numbers, the compiler infers the `Double` type.
+
+```kotlin
+val pi = 3.14 // Double
+// val one: Double = 1 // Error: type mismatch
+val oneDouble = 1.0 // Double
+```
+
+To explicitly specify the `Float` type for a value, add the suffix `f` or `F`.
+If such a value contains more than 6-7 decimal digits, it will be rounded.
+
+```kotlin
+val e = 2.7182818284 // Double
+val eFloat = 2.7182818284f // Float, actual value is 2.7182817
+```
+
+Note that unlike some other languages, there are no implicit widening conversions for numbers in Kotlin.
+For example, a function with a `Double` parameter can be called only on `Double` values, but not `Float`,
+`Int`, or other numeric values.
+
+```kotlin
+fun main() {
+    fun printDouble(d: Double) { print(d) }
+
+    val i = 1    
+    val d = 1.0
+    val f = 1.0f 
+
+    printDouble(d)
+//    printDouble(i) // Error: Type mismatch
+//    printDouble(f) // Error: Type mismatch
+}
+```
+
+To convert numeric values to different types, use [Explicit conversions](#explicit-conversions).
+
+## Literal constants
+
+There are the following kinds of literal constants for integral values:
+
+* Decimals: `123`
+    * Longs are tagged by a capital `L`: `123L`
+* Hexadecimals: `0x0F`
+* Binaries: `0b00001011`
+
+>Octal literals are not supported.
+>
+{type="note"}
+
+Kotlin also supports a conventional notation for floating-point numbers:
+
+* Doubles by default: `123.5`, `123.5e10`
+* Floats are tagged by `f` or `F`: `123.5f`
+
+You can use underscores to make number constants more readable:
+
+```kotlin
+val oneMillion = 1_000_000
+val creditCardNumber = 1234_5678_9012_3456L
+val socialSecurityNumber = 999_99_9999L
+val hexBytes = 0xFF_EC_DE_5E
+val bytes = 0b11010010_01101001_10010100_10010010
+```
+
+## Numbers representation on the JVM
+
+On the JVM platform, numbers are stored as primitive types: `int`, `double`, and so on.
+Exceptions are cases when you create a nullable number reference such as `Int?` or use generics.
+In these cases numbers are boxed in Java classes `Integer`, `Double`, and so on.
+
+Note that nullable references to the same number can be different objects:
```

Comment:
Consider replacing `can be different objects` -> `can refer to different objects`